File: 125.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        esq = 0
        dir = len(s)-1

        for i in range(len(s)):
            if s[esq].isalnum() and s[dir].isalnum():
                if s[esq].lower() != s[dir].lower():
                    return False
            elif not s[esq].isalnum():
                esq += 1
                continue
            elif not s[dir].isalnum():
                dir -= 1
                continue
            esq += 1
            dir -= 1

        return True

File: 125.cpp
class Solution
{
public:
    bool isPalindrome(string s)
    {
        int esq = 0;
        int dir = s.length() - 1;

        for (int i = 0; i < s.length(); i++)
        {
            if (isalnum(s[esq]) && isalnum(s[dir]))
            {
                if (tolower(s[esq]) != tolower(s[dir]))
                    return false;
            }
            else if (!isalnum(s[esq]))
            {
                esq += 1;
                continue;
            }
            else if (!isalnum(s[dir]))
            {
                dir -= 1;
                continue;
            }

            esq += 1;
            dir -= 1;
        }
        return true;
    }
};